src/init.jl

```diff
@@ -163,3 +165,41 @@ function __init__()
         @reexport using .UnitfulRecipes
     end
 end
+
+##################################################################
+# COV_EXCL_START
+if bool_env("PLOTS_PRECOMPILE", "true") && bool_env("JULIA_PKG_PRECOMPILE_AUTO", "true")
+    @precompile_setup begin
+        backend()  # get from `Preferences` or env, and initialize backend
```

Comment:
```suggestion

# Get from `Preferences` or env, and initialize backend
# This must always be done during precompilation.
# Otherwise, the cache will not invalidate when preferences change.
const default_backend = backend()
##################################################################
# COV_EXCL_START
if bool_env("PLOTS_PRECOMPILE", "true") && bool_env("JULIA_PKG_PRECOMPILE_AUTO", "true")
    @precompile_setup begin
```

Reply:
That seems to work, thanks. I guess it's because of the assignment to a `const` variable ?

Reply:
Wait no, I take that back.

```julia
$ JULIA_PKG_PRECOMPILE_AUTO=0 julia -e 'import Plots; Plots.set_default_backend!(; force=true)'
$ julia -e 'using Plots'  # precompiles
$ JULIA_PKG_PRECOMPILE_AUTO=0 julia -e 'import Plots; Plots.set_default_backend!(:gaston; force=true)'
$ julia -e 'using Plots'  # no precompilation